src/v/cluster/members_frontend.cc

```diff
@@ -132,4 +138,79 @@ members_frontend::recommission_node(model::node_id id) {
     }
     co_return res.value().data.error;
 }
+
+ss::future<members_frontend::get_under_replicated_reply>
+members_frontend::get_under_replicated(model::node_id id) {
+    if (id != _self()) {
+        co_return get_under_replicated_reply{
+          .under_replicated = co_await is_node_under_replicated(id),
+          .error = errc::success};
+    }
+
+    constexpr model::node_id no_leader{-1};
+    absl::flat_hash_set<model::node_id> leaders;
+    const auto& parts = _partition_manager.local().partitions();
```

Comment:
it would seem that you'd want to repeat this for each core if the goal is to find out if this node hosts some under replicated partitions?